src/service/filtering/filtering_processing.py

```diff
@@ -0,0 +1,168 @@
+) -> FilterSet:
+    """
+    Append filters to a FilterSet based on the provided filter query and column definitions.
+
+    fs - The FilterSet to which filters will be appended.
+    filter_query - The filter query parameters containing field names and filter strings.
+    columns - The columns to which the filters apply, represented as a dictionary
+        mapping field names to AttributesColumn objects.
+    trans_field_func - A function to transform the field name to valid column name in the filter query
+        before applying it to the FilterSet. Default is None.
+
+    """
+    for field, querystring in filter_query.items():
+        if field not in columns:
+            raise errors.IllegalParameterError(f"No such filter field: {field}")
+        column = columns[field]
+        minlen = analyzers.get_minimum_query_length(column.filter_strategy)
+        if minlen and len(querystring) < minlen:
+            raise errors.IllegalParameterError(
+                f"Filter field '{field}' requires a minimum query length of {minlen}")
+
+        # Translates the query field name to an available column name if necessary.
+        # For example, translates query field name '1' to a valid column name 'col_1_val'.
+        field = trans_field_func(field) if trans_field_func else field
+
+        fs.append(
+            field,
+            column.type,
+            querystring,
+            analyzers.get_analyzer(column.filter_strategy),
+            column.filter_strategy
+        )
+    return fs
+
+
+async def get_filters(
+        r: Request,
+        arango_coll: str,
+        coll_id: str,
+        load_ver: str,
+        load_ver_override: bool,
+        data_product: str,
+        columns: list[col_models.AttributesColumn],
+        view_name: str = None,
+        count: bool = False,
+        sort_on: str = None,
+        sort_desc: bool = False,
+        filter_conjunction: bool = True,
+        match_spec: SubsetSpecification = None,
+        selection_spec: SubsetSpecification = None,
+        keep: dict[str, set[col_models.ColumnType]] = None,
+        keep_filter_nulls: bool = False,
+        skip: int = 0,
+        limit: int = 1000,
+        start_after: str = None,
+        trans_field_func: Callable[[str], str] = None,
+) -> FilterSet:
+    """
+    Constructs a FilterSet and applies filters using the provided filter query and columns.
+
+    Requires a 'get_columns_func' to generate columns, represented as a dictionary mapping column name/ID to
+    columnar_attribs_common_models.AttributesColumn objects.
+
+    r - The request.
+    arango_coll - The name of the Arango collection.
+    coll_id - The collection ID.
+    load_ver - The load version.
+    load_ver_override - Whether or not to override the load version.
+    data_product - The data product id.
+    columns - list of columnar_attribs_common_models.AttributesColumn objects to be filtered on.
+    view_name - The name of the view to use for filtering.
```

Comment:
Just some minor clarification

```suggestion
    arango_coll - The name of the Arango collection to filter on.
    coll_id - The KBase Collection ID for the Collection that will be filtered.
    load_ver - The load version of the KBase Collection.
    load_ver_override - Whether or not the load version was overridden by a collections
        admin.
    data_product - The ID of the data product being filtered.
    columns - the column definitions for the data product and KBase Collection.
    view_name - The name of the ArangoSearch view to use for filtering, if any.
        A view must be supplied if any filter query parameters are passed in the request.
```